### PR TITLE
Add kube-requests-checker tool and doc.

### PR DESCRIPTION
**Pull request by agmsb**
Adding the initial iteration of the `kube-requests-checker` tool.

### CODE REVIEW

File: state-of-k8s-cost-optimization/kube-requests-checker.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+
+# Setup variables
+
+CLI_NAME=${0}
+OPTIONS=$(getopt -o n:l:ih --long namespace:,level:,include-kube-system,help -- "$@")
+EXCLUSIONS="--field-selector metadata.namespace!=kube-system"
```
Comment (minherz) [inline state-of-k8s-cost-optimization/kube-requests-checker.sh:23]: nit: consider defining these variables as `local`

File: state-of-k8s-cost-optimization/kube-requests-checker.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    json=$(kubectl get pods ${NAMESPACE} ${EXCLUSIONS} \
+            -o json | \
+            jq '[.items[] | {kind: .kind, namespace: .metadata.namespace, name: .metadata.name, podQoS: .status.qosClass, containers: [ .spec.containers[] | {name: .name, cpuRequest: .resources.requests.cpu, cpuLimit: .resources.limits.cpu, memRequest: .resources.requests.memory, memLimit: .resources.limits.memory}]}]')
+else
+    if [ "${LEVEL}" = "" ] || [ "${LEVEL}" = "controller" ]; then
+        LEVEL="controller"
+    else
+       echo "Invalid option: '${LEVEL}' is not valid. Must be 'pod' or 'controller'."
+       exit 1
+    fi
```
Comment (minherz) [inline state-of-k8s-cost-optimization/kube-requests-checker.sh:105]: nit: maybe refactoring it to `if elif else` instead to improve script readability?

**Review by minherz — APPROVED** [review]

**Review by bourgeoisor — CHANGES_REQUESTED** [review]
Could you move this here instead?
`<repo-root>/cost-optimization/state-of-k8s-cost-optimization/`

File: state-of-k8s-cost-optimization/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# kube-requests-checker
```
Comment (bourgeoisor) [inline state-of-k8s-cost-optimization/README.md:1]: In the README, could we also point to the live c.g.c doc that this sample is used in?

Comment (agmsb) [reply state-of-k8s-cost-optimization/README.md:1]: This won't be in CGC doc, but rather linked in a CGC blog going out next week. Once the blog is live can I open a separate PR and update the README?

File: state-of-k8s-cost-optimization/kube-requests-checker.sh (hunk 1 above)
Comment (agmsb) [reply state-of-k8s-cost-optimization/kube-requests-checker.sh]: can i do this outside of a bash function?

**Comment by agmsb** [comment]
> Could you move this here instead? `<repo-root>/cost-optimization/state-of-k8s-cost-optimization/`

This is done. Working on other nits.

File: state-of-k8s-cost-optimization/kube-requests-checker.sh (hunk 2 above)
Comment (agmsb) [reply state-of-k8s-cost-optimization/kube-requests-checker.sh]: This is done. Thanks!

**Review by bourgeoisor — APPROVED** [review]
LGTM

**Pull request merged**